src/proxyp/Elements.cc

```diff
@@ -11,28 +11,53 @@
 #include "proxyp/Elements.h"
 #include "sbuf/Stream.h"
 
+#include <algorithm>
 #include <limits>
+#include <map>
 
-const ProxyProtocol::FieldMap ProxyProtocol::PseudoHeaderFields = {
-    { SBuf(":version"), ProxyProtocol::Two::htPseudoVersion },
-    { SBuf(":command"), ProxyProtocol::Two::htPseudoCommand },
-    { SBuf(":src_addr"), ProxyProtocol::Two::htPseudoSrcAddr },
-    { SBuf(":dst_addr"), ProxyProtocol::Two::htPseudoDstAddr },
-    { SBuf(":src_port"), ProxyProtocol::Two::htPseudoSrcPort },
-    { SBuf(":dst_port"), ProxyProtocol::Two::htPseudoDstPort }
+namespace ProxyProtocol {
+namespace Two {
+
+/// a mapping between pseudo header names and ids
+typedef std::map<SBuf, FieldType> FieldMap;
+static const FieldMap PseudoHeaderFields = {
+    { SBuf(":version"), htPseudoVersion },
+    { SBuf(":command"), htPseudoCommand },
+    { SBuf(":src_addr"), htPseudoSrcAddr },
+    { SBuf(":dst_addr"), htPseudoDstAddr },
+    { SBuf(":src_port"), htPseudoSrcPort },
+    { SBuf(":dst_port"), htPseudoDstPort }
 };
 
-namespace ProxyProtocol {
+} // namespace Two
+
 static Two::FieldType NameToFieldType(const SBuf &);
 static Two::FieldType IntegerToFieldType(const SBuf &);
+
 } // namespace ProxyProtocol
 
+
+const SBuf &
+ProxyProtocol::PseudoFieldTypeToFieldName(const Two::FieldType fieldType)
+{
+    // reverse PseudoHeaderFields so that we can look up names by FieldType
+    typedef std::vector<SBuf> PsuedoFieldNames;
```

Comment:
This typedef is unnecessary code.

Reply:
Left as is because naming non-trivial types is good practice: Naming makes some compiler errors clearer, often simplifies working with a debugger, may reduce future code changes, and may reduce instances where copy-pasting folks incorrectly duplicate types because they forget to name them (in contexts where the type is actually spelled out multiple times).

I will refactor to remove this declaration if you insist on this change. Do you insist on this change?

Reply:
good != best. And the nature of its goodness is debatable. If it were a long function or a deeply nested template definition - sure typedef has merits. This is not one of those cases.

FYI, there have been some serious conversations starting this past year in Debian about large software (eg Firefox, Apache, Bind, Chrome) becoming non-compilable due to symbol table bloat slowing the compiler and linker down. You may recall some times in the past when Squid itself hit these issues in the BuildFarm VMs. To void spreading that further IMO we should only typedef when it is actively useful. With C++11 theoretically we should not need to use it at all.

Reply:
I am interpreting your comment as "yes, I insist on this change" answer. Please let me know if have misinterpreted it.

Reply:
@eduard-bagdasaryan, after fixing runtime errors, please remove this typedef to address this change request.

Reply:
Removed at 87f38f.